docs/mkdocs/documentation/ordered_upgrade.md

```diff
@@ -0,0 +1,72 @@
+# Ordered upgrade of kernel module without reboot
+
+Sometimes customer needs to update the kernel module code for a specific version,
+due to new feature added to the kernel module's code, or a bug fix was pushed
+into the source of the kernel module.
+The upgrade should not require node reboot.
+It should be executed sequentially, one node at a time,
+so as not to disturb the workload of the cluster. The order of the 
+upgrade is managed by the admin/user.
+
+## Prerequisite
+
+In order to successfully upgrade a kernel module to a new version, the currently loaded kernel module
+should be unloaded, and a new kernel module file should be loaded. That means that all user-mode 
+application/workloads accessing kernel module should be deleted prior to attempting to unload kernel 
+module. In k8s cluster there are 2 type of workloads that can access KMM kernel module:
+1. user/customer workload: application running on k8s cluster that is accessing kernel module. It is
+   deployed by the customer/admin, and it is customer/admin responsibility to make sure that the workload
+   is not running on the node prior to kernel module unloading, and to make sure that the workload
+   is back running on the node once the new kernel module version has been loaded
+2. device plugin: managed by KMM operator, that will also manage the unloading/loading of the device plugin
+   on a node being upgraded
```

Comment:
2 scenarious:
1) dependency is defined by multiple Modules, in which case it is managed by the customer's operator. In that case customer operator will have to manage the correct sequence by changing the correct labels.
2) dependency is define by depmod or by order in one module. In that case it will work as usual, since removal adding of the kernel modules will be done in the correct order by modprobe

Reply:
I just think we should mention it in the doc